src/main.rs

```diff
@@ -1,12 +1,83 @@
-use std::io::Read;
+use jotdown::{html, Parser};
+use std::{
+    ffi::OsString,
+    fs::{self, File},
+    io::{self, BufWriter},
+    process::exit,
+};
+
+struct App {
+    input: Option<OsString>,
+    output: Option<OsString>,
+    format: OsString,
+}
+
+impl Default for App {
+    fn default() -> Self {
+        Self {
+            input: None,
+            output: None,
+            format: "html".into(),
+        }
+    }
+}
+
+fn parse_args() -> App {
+    let mut app = App::default();
+
+    let mut args = std::env::args_os().skip(1).peekable();
+
+    while let Some(arg) = args.next() {
+        match (arg.to_string_lossy().as_ref(), args.peek()) {
+            ("-h" | "--help", _) => {
+                eprint!("{}", include_str!("./help.txt"));
+                exit(0);
+            }
+            ("-v" | "--version", _) => {
+                eprintln!("jotdown v{}", env!("CARGO_PKG_VERSION"));
+                exit(0);
+            }
+            ("-o" | "--output", Some(o)) => {
+                app.output = Some(o.into());
+                args.next();
```

Comment:
similarly here, if no filename is given after "-o", it just continues to use stdout instead

Reply:
Both this and the previous comment will be easily resolved by adding another match arm. I'll get straight to it